### PR TITLE
Gold 290a : Add Sanitization of TTLMap Entry

### PR DESCRIPTION
**Pull request by abdulazeem-tk4vr**
Addresses sanitization of map entry.
https://linear.app/shm/issue/SHARD-461/blacklisting-of-ips-in-the-json-rpc-server-which-have-caused-timeouts

### CODE REVIEW

**Review by jintukumardas — DISMISSED** [review]

File: test/unit/testingNodeIpPort.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import net from 'net'
```
Comment (aniketdivekar) [inline test/unit/testingNodeIpPort.ts]: This is not the correct way to write unit tests. Please take shardeum, shardus-core as example and write tests accordingly.

Comment (jintukumardas) [reply test/unit/testingNodeIpPort.ts]: +,  I think the filename name should be like this: {FILENAME}.test.ts. You can ignore the "testing" from the filename.

**Review by jintukumardas — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -62,13 +63,30 @@
...
+    ],
+    "moduleFileExtensions": [
+      "ts",
+      "js",
+      "json"
+    ],
+    "collectCoverageFrom": [
+      "src/**/*.{ts,js}"
+    ]
   }
```
Comment (chrypnotoad) [inline package.json]: We keep this in a separate `jest.config.js` file in our other repos see shardeum https://github.com/shardeum/shardeum/blob/dev/jest.config.js

File: package.json [hunk 3]
```diff
@@ -53,6 +53,7 @@
     "@types/cors": "2.8.13",
     "@types/express": "4.17.15",
     "@types/hapi__sntp": "^3.1.4",
+    "@types/jest": "^29.5.14",
```
Comment (chrypnotoad) [inline package.json]: remove `^` version should be locked

File: package.json [hunk 4]
```diff
@@ -62,13 +63,30 @@
     "eslint-plugin-no-unsanitized": "4.0.2",
     "eslint-plugin-security": "1.5.0",
     "eslint-plugin-xss": "0.1.12",
-    "nodemon": "2.0.22"
+    "jest": "^29.7.0",
+    "nodemon": "2.0.22",
+    "ts-jest": "^29.2.5"
```
Comment (chrypnotoad) [inline package.json]: remove `^` version should be locked

**Review by chrypnotoad — COMMENTED** [review]
few minor comments on package.json. Build is working and tests are not failing so looks good. no hard blockers

**Review by aniketdivekar — APPROVED** [review]

**Review by jintukumardas — APPROVED** [review]

**Review by mgthuramoemyint — APPROVED** [review]
approved

**Pull request merged**